R-package/R/lightgbm.R

```diff
@@ -120,7 +120,7 @@ lightgbm <- function(data,
                      verbose = 1L,
                      eval_freq = 1L,
                      early_stopping_rounds = NULL,
-                     save_name = "lightgbm.model",
+                     save_name = NULL,
```

Comment:
This change would break existing code that relies on this behavior _silently_. Code relying on this behavior might fail with a "file not found" error after `lightgbm::lightgbm()` has been called, resulting in possibly-significant time spent debugging to figure out that `{lightgbm}`'s default behavior has changed.

I agree with the argument in the pull request description that model `lightgbm::lightgbm()` shouldn't be responsible for persisting models to disk. I support just removing the argument `save_name` from `lightgbm()` entirely.  I think we should make that change right now, in this PR, without waiting for a round of deprecation warnings, because:

1.  it's already been almost 4 months since the most recent LightGBM release ([v3.3.1](https://github.com/microsoft/LightGBM/releases/tag/v3.3.1), excluding the special v3.3.2 that only had one small patch) and I expect it to be at least 2 months until the next, 4.0.0, release
2. this breaking change would result in a loud training-time error, which I consider much less expensive than than a silent breaking change or than one that happens at scoring time with a deployed model

@StrikerRUS @Laurae2 , what do you think?

Reply:
I agree with both points above. Thanks a lot for the detailed explanation!

Reply:
Removed.